### ISSUE TITLE
Behavior when documenting a __slots__ attribute is unexpected

### ISSUE THREAD

**Issue by 13steinj** [sphinx-doc/sphinx#4678]
Subject: Behavior when documenting a __slots__ attribute is unexpected

### Problem
When using autodoc to document a class with a `__slots__` attribute, some unexpected behavior occurs. Slots _can_ be members of both the class and the object, however this only occurs if they are initialized in the class definition, however they do not have to be, and in this case are instance variables only, if they are even set, not class members. Furthermore while they do exist on the class if one tries to use `getattr`, it specifically defines it as a special type used only for slots:`<member 'somevar' of 'foo' objects>`.

In my project (which is private while in development, however, I've made a public repo and branch so others can see the issue [here](https://github.com/13steinj/sphinx-issue-examples/tree/autodoccedslots), If I want to document all class members, slots also are listed as class members, even when not initialized. The project is quite large and as such I have both human made documentation, but other than that let the extensive docstrings and autodoc suffice.

But there's no way to indicate that an attribute is an object member slot. Furthermore we can't document these attributes with any useful information, because they are slots, not actual attributes on the class. So if we do use the `ivar` field syntax, then it gets listed twice, and linked to, which does not make sense.

#### Procedure to reproduce the problem
Have a slots attribute with variables that are only initialized within the class itself, but are either not documented because they are documented in other ways, or are not meant to be documented because while the slot exists, no method in the class definition actually sets it in the object, so it can't be reasonably documented, even if the user wants to.
#### Error logs / results
https://github.com/13steinj/sphinx-issue-examples/tree/autodoccedslots
#### Expected results
<Describe what to actually do>
I propose that the slots of a class are to be parsed visually with their own rst marker, ex `:slots:`, if the user wishes. However otherwise they aren't to be shown as members if they aren't initialized, undoced or not, because on the class they do not provide useful information-- especially if used only internally. If the `__slots__` attribute itself is marked up to be displayed, this new rst marker's output would be put underneath it. How the html would be drawn up for this, I don't know, I'm not a designer. 

### Reproducible project / your project
https://github.com/13steinj/sphinx-issue-examples/tree/autodoccedslots
### Environment info
- OS: Ubuntu
- Python version:2, however the issue would seem independent of python version
- Sphinx version: 1.6.5, however a cursory look at the change log doesn't indicate this resolved
- stdlib SimpleHTTPServer, any browser

**Comment by cjw296** [comment]
`__slots__` also seem to break `:member-order:` and prevent auto-doc seeing class attributes.
Should I file separate bugs for both of those?

**Comment by 13steinj** [comment]
@cjw296 I hold no power in this repo, but imo probably-- both are separate issues. Or a major issue with a checklist of what I described and what you did. Again though, I'm not one to say.

**Comment by tk0miya** [comment]
@cjw296 Could you file a new issue please?

**Comment by tk0miya** [comment]
Note: As a workaround, you can give a comment for members on `__init__()` method.
```
def __init__(self, *a):
    #: details of a
    self.somevar = a
```

**Comment by tk0miya** [comment]
Fixed by #6388.
Thank you for reporting.
